env/common/templates/galaxy/files/xsede_report.py.j2

```diff
@@ -0,0 +1,179 @@
+    if debug:
+        print('Executed:')
+        print(cur.query)
+    print('Query returned %d rows' % cur.rowcount)
+    rows = cur.fetchall()
+    cur.close()
+    return rows
+
+
+def submit(row, args):
+    xsede_job_id, xsede_submit_time, username, xsede_resource = format_data(row[0], row[1], row[2], row[3], args)
+    xsede_url = "https://xsede-xdcdb-api.xsede.org/gateway/v2/job_attributes"
+    xsede_data = {
+        "gatewayuser": username,
+        "xsederesourcename": xsede_resource,
+        "jobid": xsede_job_id,
+        "submittime": xsede_submit_time,
+        "apikey": GALAXY_XSEDE_APIKEY
+    }
+    if args.debug:
+        xsede_data["debug"] = True
+    return requests.post(xsede_url, data=xsede_data)
+
+
+def format_data(job_id, submit_time, username, resource, args):
+    xsede_submit_time = submit_time.strftime("%d-%m-%y %H:%M %z")
+    xsede_job_id = args.galaxy_name + "_" + str(job_id)
+    for key in DESTINATION_MAPPING:
+        if resource.startswith(key):
+            xsede_resource = DESTINATION_MAPPING[key]
+            break
+    return xsede_job_id, xsede_submit_time, username, xsede_resource
+
+
+def get_current_offset(given_offset):
+    """
+    Calculate current query offset by adding the value in
+    offset file to the offset given as an argument.
+    """
+    if not os.path.exists(OFFSET_FILE_PATH):
+        current_offset = given_offset
+    else:
+        with open(OFFSET_FILE_PATH, "r") as f:
+            line = f.readline()
+            offset = 0
+            if line != "":
+                try:
+                    offset = int(line)
+                except Exception:
+                    print("Unable to parse offset file, ignoring.")
+            current_offset = offset + given_offset
+    return current_offset
+
+
+def write_current_offset(current_offset):
+    """
+    Store the current offset in the file.
+    """
+    with open(OFFSET_FILE_PATH, "w") as f:
+        f.write(str(current_offset))
+
+
+def main():
+    args = parse_arguments()
+    rows = query(**vars(args))
+    print("Sending data to XSEDE.")
+    reported_jobs = 0
+    for row in rows:
+        response = None
+        try:
+            response = submit(row, args)
+            assert response.status_code in [200]
+            print("OK " + str(format_data(row[0], row[1], row[2], row[3], args)))
+            reported_jobs += 1
+        except Exception as e:
+            print("Breaking the submission process and increasing offset.")
+            write_current_offset(args.current_offset)
+            print("Submission of the object below failed:")
+            print(response.headers)
+            print(response.text)
+            print("Full script options:")
+            for i in vars(args).items():
+                print('%s: %s' % i)
+            raise e
+    print("Succesfully reported %s jobs out of %s." % (reported_jobs, len(rows)))
+    if reported_jobs == len(rows):
+        write_current_offset(0)
```

Comment:
Is it possible for some to succeed and some to fail, resulting in double submission?

Reply:
Per info from them that is fine and will result in a single entry. This line is just a sanity check for couple of initial runs, then we can just drop it along with the printing.